src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -783,11 +786,15 @@ private EC2AbstractSlave provisionSpot(TaskListener listener) throws AmazonClien
     }
 
     protected EC2OndemandSlave newOndemandSlave(Instance inst) throws FormException, IOException {
-        return new EC2OndemandSlave(inst.getInstanceId(), description, remoteFS, getNumExecutors(), labels, mode, initScript, tmpDir, remoteAdmin, jvmopts, stopOnTerminate, idleTerminationMinutes, inst.getPublicDnsName(), inst.getPrivateDnsName(), EC2Tag.fromAmazonTags(inst.getTags()), parent.name, usePrivateDnsName, useDedicatedTenancy, getLaunchTimeout(), amiType);
+        EC2OndemandSlave slave= new EC2OndemandSlave(inst.getInstanceId(), description, remoteFS, getNumExecutors(), labels, mode, initScript, tmpDir, remoteAdmin, jvmopts, stopOnTerminate, idleTerminationMinutes, inst.getPublicDnsName(), inst.getPrivateDnsName(), EC2Tag.fromAmazonTags(inst.getTags()), parent.name, usePrivateDnsName, useDedicatedTenancy, getLaunchTimeout(), amiType);
+        slave.setOnetimeUse(this.onetimeUse);
+        return slave;
```

Comment:
Seems like it would be smarter to:
1. Refactor the slave constructor to use a new EC2SlaveOptions class (rather than 20 different variables)
2. Add onetimeUse as another option